api4/user.go

```diff
@@ -123,16 +123,9 @@ func createUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	var ruser *model.User
 	var err *model.AppError
 	if tokenId != "" {
-		token, nErr := c.App.Srv().Store.Token().GetByToken(tokenId)
-		if nErr != nil {
-			var status int
-			switch nErr.(type) {
-			case *store.ErrNotFound:
-				status = http.StatusNotFound
-			default:
-				status = http.StatusInternalServerError
-			}
-			c.Err = model.NewAppError("CreateUserWithToken", "api.user.create_user.signup_link_invalid.app_error", nil, nErr.Error(), status)
+		token, appErr := c.App.GetCreateUserToken(tokenId)
```

Comment:
This looks like it is just getting the token. Let's name it `GetTokenByUserId`

Reply:
Hey @agnivade, I'm bit out of domain knowledge, shouldn't we better name it `GetTokenByTokenId` then?

Reply:
My bad there. It's not the userID. `GetTokenById` should be fine. Thanks @dizlv.